src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/PersonalisationProvider.java

```diff
@@ -358,6 +358,18 @@ public Map<String, String> getRespondToCostsApplicationNumber(AsylumCase asylumC
             .build();
     }
 
+    public Map<String, String> getAdditionalEvidenceApplicationNumber(AsylumCase asylumCase) {
+        DynamicList applyForCostsDynamicList = asylumCase.read(ADD_EVIDENCE_FOR_COSTS_LIST, DynamicList.class)
+            .orElseThrow(() -> new IllegalStateException("dynamicList is not present"));
+
+        String applicationNumber = applyForCostsDynamicList.getValue().getLabel().split(",")[0].replaceAll("[^0-9]", "");
+
+        return ImmutableMap
+            .<String, String>builder()
+            .put("applicationId", applicationNumber)
+            .build();
```

Comment:
This part could be extracted into separate method to avoid code duplication(we have the same lines in getRespondToCostsApplicationNumber())